### DISCUSSION TITLE
Auto Resevered based Ping checking

### DISCUSSION THREAD

**Discussion by MeirPeleg** [nautobot/nautobot#4356] [Ideas]
Any idea how can we add plugin that will ping IP of selected prefixed and then will change status automatically?

**Comment by jdrew82** [comment]
You could write an SSoT App that performs the ping sweep of your network, loads the prefixes that respond, and then load what's in Nautobot and it'll figure out what needs to be added, updated, removed, etc.

**Comment by zackt25** [comment]
You could also make a job that goes through your devices pings or connects then updates either extra field on device or maybe status. You could use nornir/netmiko or even just subcommand module to ping.

**Reply by MeirPeleg** [reply]
What lanauges /  Code can I prepare this code? python only?

**Reply by jdrew82** [reply]
Jobs are written in Python, yes.
